micro-frontends/webpack.config.js

```diff
@@ -0,0 +1,136 @@
+
+module.exports = {
+  resolve: {
+    extensions: [".tsx", ".ts", ".jsx", ".js", ".json"],
+    alias: {
+      react: path.resolve(__dirname, "./src/globalReact.js"),
+      'react-dom': path.resolve(__dirname, "./src/globalReactDom.js"),
+    }
+  },
+  entry: {
+    ipd: "./src/ipd/index.js",
+    mfe_polyfills_angular_1_4: "./src/polyfill.js",
+  },
+  output: {
+    path: path.resolve(__dirname, "../ui/app/micro-frontends-dist"),
+    filename: "[name].min.js",
+    clean: true,
+  },
+  devServer: {},
+  plugins: [
+    new cssExtract({
+      filename: "mfe-styles.css",
+    }),
+    new ModuleFederationPlugin({
+      name: "bahmni_mfe_host",
+      filename: "remoteEntry.js",
+      remotes: {
+        "@openmrs-mf/ipd": dynamicRemote('bahmni_ipd', 'ipd'),
+      },
+      exposes: {},
+      shared: {
+        "carbon-components-react": {
+          singleton: true,
+          eager: true,
+          requiredVersion: dependencies["carbon-components-react"],
+        },
+        "carbon-components": {
+          singleton: true,
+          eager: true,
+          requiredVersion: dependencies["carbon-components"],
+        },
+        "bahmni-carbon-ui": {
+          singleton: true,
+          eager: true,
+          requiredVersion: dependencies["bahmni-carbon-ui"],
+        },
+        react: {
+          singleton: true,
+          eager: true,
+          requiredVersion: dependencies.react,
+        },
+        "react-dom": {
+          singleton: true,
+          eager: true,
+          requiredVersion: dependencies["react-dom"],
+        },
+      },
+    }),
+  ],
+  module: {
+    // noParse: ['react'],
+    rules: [
+      {
+        test: /\.m?js/,
+        type: "javascript/auto",
+        resolve: {
+          fullySpecified: false,
+        },
+      },
+      {
+        test: /\.(ts|tsx|js|jsx)$/,
+        exclude: /node_modules/,
+        use: {
+          loader: "babel-loader",
+        },
+      },
+      {
+        test: /\.css$/i,
+        use: [cssExtract.loader, "css-loader"],
+      },
+      {
+        test: /\.s[ac]ss$/i,
+        use: [cssExtract.loader, "css-loader", "sass-loader"],
+      },
+    ],
+  },
+  externals: {
+    react: "React", // Exclude react from the bundled output
+    "react-dom": "ReactDOM", // Exclude react-dom from the bundled output
+  },
+};
+
+
+/**
+ * An alternative to providing build time URLs
+ * We need to do this string promise stuff because we need to resolve the host at run-time.
+ * This is the way, as documented here: https://webpack.js.org/concepts/module-federation/#promise-based-dynamic-remotes
+ * 
+ * @param {string} name The name of the remote, as given in it's ModuleFederationPlugin configuration
+ * @param {string} subPath The sub-path of the remote, as set-up in the proxy configuration
+ * 
+ * @returns {string} A string that can be evaluated to a promise that resolves to the remote
+ */
+function dynamicRemote(name, subPath) {
```

Comment:
Can you explain the rationale behind using `dynamicRemote` instead of `remoteProxiedAtHostDomain`?